core/log/batch_logger.cpp

```diff
@@ -13,6 +13,53 @@
 namespace gko {
 namespace batch {
 namespace log {
+// namespace detail {
```

Comment:
what's going on with this?

Reply:
maybe the header was missing some `inline`? Still it should be moved into the core.